[developer docs] Improve instructions to setup Pulsar development in IntelliJ (#8892)
apache/pulsar: README.md

## Build Pulsar

Requirements:
 * Java 8 JDK (for building Pulsar)

>    * When building on newer than Java 8, the resulting artifacts are not compatible with Java 8 runtime because of issues such as https://github.com/apache/pulsar/issues/8445.

Reviewer: ```suggestion
   * When building Pulsar on a higher version (higher than Java 8), the resulting artifacts are not compatible with Java 8 runtime because of some issues, such as [issue 8445](https://github.com/apache/pulsar/issues/8445).
```

5. Install the lombok plugin in intellij.

#### Further configuration in IntelliJ 

* When working on the Pulsar core modules in IntelliJ, reduce the number of active projects in IntelliJ to speed up IDE actions and reduce unrelated IDE warnings.
  * In IntelliJ's Maven UI's tree view under "Profiles"

>     * Activate "core-modules" maven profile

Reviewer: use "Maven" instead of ""maven.Please check through the whole updates.

Author: done

    * De-activate "main" maven profile

>     * Run "Reload All Maven Projects" action from the Maven UI toolbar. You can also find the action by name in the IntelliJ "Search Everywhere" window that gets activated by pressing the shift key twice.

Reviewer: ```suggestion
    * Run the "Reload All Maven Projects" action from the Maven UI toolbar. You can also find the action by the name in the IntelliJ "Search Everywhere" window that gets activated by pressing the **Shift** key twice.
```

> * Run "Generate Sources and Update Folders For All Projects" from the Maven UI toolbar. You can also find the action by name in the IntelliJ "Search Everywhere" window that gets activated by pressing the shift key twice. Running the action takes about 10 minutes for all projects. This is faster when the "core-modules" profile is the only active profile.

Reviewer: ```suggestion
* Run the "Generate Sources and Update Folders For All Projects" action from the Maven UI toolbar. You can also find the action by the name in the IntelliJ "Search Everywhere" window that gets activated by pressing the **Shift** key twice. Running the action takes about 10 minutes for all projects. This is faster when the "core-modules" profile is the only active profile.
```

#### IntelliJ usage tips

> * In the case of compilation errors with missing Protobuf classes, make sure to run "Generate Sources and Update Folders For All Projects" action.

Reviewer: ```suggestion
* In the case of compilation errors with missing Protobuf classes, ensure to run the "Generate Sources and Update Folders For All Projects" action.
```

* All of the Pulsar source code doesn't compile properly in IntelliJ and there are compilation errors.
  * Use the "core-modules" profile if working on the Pulsar core modules since the source code for those modules can be compiled in IntelliJ.

>   * Sometimes it might help to mark a specific project ignored in IntelliJ Maven UI by right clicking the project name and choosing "Ignore Projects".

Reviewer: ```suggestion
  * Sometimes it might help to mark a specific project ignored in IntelliJ Maven UI by right-clicking the project name and select **Ignore Projects** from the drop-down list.
```

>   * Because of the compilation issues, it's not currently possible to run all unit tests directly from the IDE. As a workaround, individual test classes can be run from the command line using "mvn test -Dtest=TestClassName" command.

Reviewer: ```suggestion
  * Currently, it is impossible to run all unit tests directly from the IDE because of the compilation issues. As a workaround, individual test classes can be run by using the `mvn test -Dtest=TestClassName` command.
```

Author: Thank you for the review and suggestions @Huanli-Meng . I made a slight change to the wording since "impossible" is a bit too strong word in this context. I used "not always possible" instead. Please review the latest changes.
